pyQuARC/code/custom_checker.py

```diff
@@ -137,24 +177,27 @@ def run(
 
         invalid_values = []
         validity = None
-        for arg in args:
-            function_args = [*arg]
-            function_args.extend(
-                [
-                    extra_arg
-                    for extra_arg in [relation, *external_data, external_relation]
-                    if extra_arg
-                ]
-            )
-            func_return = func(*function_args)
-            valid = func_return["valid"]  # can be True, False or None
-            if valid is not None:
-                if valid:
-                    validity = validity or (validity is None)
-                else:
-                    if "value" in func_return:
-                        invalid_values.append(func_return["value"])
-                    validity = False
+
+        # Process arguments using multithreading
+        with ThreadPoolExecutor() as executor:
+            future_results = []
+            for arg in args:
+                future = executor.submit(
+                    self._process_argument,
+                    arg,
+                    func,
+                    relation,
+                    external_data,
+                    external_relation,
+                    invalid_values,
+                    validity,
+                )
+                future_results.append(future)
+
+            # Retrieve results from futures
+            for future in future_results:
+                invalid_values, validity = future.result()
```

Comment:
Wouldn't sub-threading be an issue?

Reply:
Did manual testing with a curated [list](https://docs.google.com/spreadsheets/d/1A1ivCqU0KPMv3RZ20pDW7Aga27JSZ4j9yCAP7KMA4sY/edit#gid=0) of concept ids, no issue at all.